Common/src/toolboxes/C1DInterpolation.cpp

```diff
@@ -0,0 +1,202 @@
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+
+#include "../../include/toolboxes/C1DInterpolation.hpp"
+
+/*--- Function for evaluating the value for akima spline ---*/
+su2double CAkimaInterpolation::EvaluateSpline(su2double Point_Interp){
+    Point_Match = true;
+    int i =0, j=this->n-1;
+
+    while (j-i>1){ 
+        int m=(i+j) / 2 ; 
+            if (Point_Interp>this->x[m]) {i=m;}
+            else {j=m;} 
+    }
+
+    su2double h=Point_Interp-this->x[i] ;
+    return this->y[i]+h*(this->b[i]+h*(this->c[i]+h*this->d[i])) ;
+}
+
+/*--- Function for evaluating the value for linear 'spline' ---*/
+su2double CLinearInterpolation::EvaluateSpline(su2double Point_Interp){
+    int size = this->x.size();
+
+    for (int i=0;i<size-1;i++){
+        if(Point_Interp>=this->x[i] && Point_Interp<=this->x[i+1]){
+            this->Point_Match = true;
+            return (Point_Interp-this->x[i])*dydx[i]+this->y[i];}
+    }
+    return 0;
+}
+
+/*--- Function for setting the cofficients for linear 'spline' ---*/
+void CLinearInterpolation::SetSpline(vector<su2double> &X, vector<su2double> &Data){
+    int n = X.size();
+    vector<su2double> h (n-1);
+    this->x.resize(n);
+    this->y.resize(n);
+    this->dydx.resize(n-1);
+
+    for (int i=0; i<n ; i++){
+        this->x[i] = X[i];
+        this->y[i] = Data[i];
+    }
+
+    for (int i=0; i<n-1;i ++){h[i]=X[i+1]-X[i];}
+    for (int i=0; i<n-1;i++) {this->dydx[i]=(Data[i+1]-Data[i]) / h [i];}
+}
+
+
+/*--- Function for setting the cofficients for akima spline ---*/
+void CAkimaInterpolation::SetSpline (vector<su2double> &X,vector<su2double> &Data){
+
+    int n = X.size();
+    vector<su2double> h (n-1);
+    vector<su2double> p (n-1);
+
+    /*---calculating finite differences (h) and gradients (p) ---*/
+    for (int i=0; i<n-1;i ++){h[i]=X[i+1]-X[i];}
+    for (int i=0; i<n-1;i++) {p[i]=(Data[i+1]-Data[i])/h[i];} 
+
+    /*---b,c,d are the akima spline's cofficient for the cubic equation---*/
+    this->x.resize(n);
+    this->y.resize(n);
+    this->b.resize(n);
+    this->c.resize(n-1);
+    this->d.resize(n-1);
+    this->n=n; 
+
+    for (int i=0; i<n ; i++){
+        this->x[i]=X[i]; 
+        this->y[i]=Data[i];
+    }
+
+    this->b[0] = p[0] ; 
+    this->b[1] =(p[0]+ p[1])/2 ;
+    this->b[n-1]=p[n-2]; 
+    this->b [n-2]=(p[n-2]+p[n-3])/2;
+
+    for (int i =2; i<n-2; i ++){
+        su2double w1=fabs(p[i+1]-p[i]) , w2=fabs(p[i-1]-p[i-2]);
+
+        if (w1+w2==0) {
+            this->b[i] = (p[i-1]+p[i])/2 ;
+        }
```

Comment:
it is usually recommended not to compare floating point values with `==`, it is better to do use `<` with a small value

Reply:
that makes sense.